Server/src/bmp/parseBMP.cpp

```diff
@@ -769,22 +769,29 @@ void parseBMP::handleInitMsg(int sock, MsgBusInterface::obj_router &r_entry) {
          */
         switch (initMsg.type) {
             case INIT_TYPE_FREE_FORM_STRING :
-                memcpy(r_entry.initiate_data, initMsg.info, initMsg.len);
+                infoLen = sizeof(r_entry.initiate_data) < initMsg.len ? sizeof(r_entry.initiate_data) : initMsg.len;
```

Comment:
When using either sizeof(...) or initMsg.len, we need to make sure that the last byte of the buffer is a null terminating byte for printf/sprintf.   How about the below for each of them?

```c
infoLen = sizeof(r_entry.initiate_data) < (initMsg.len -1) ? (sizeof(r_entry.initiate_data) - 1) : initMsg.len;
```